pkg/controller/master/upgradelog/controller.go

```diff
@@ -581,6 +587,54 @@ func (h *handler) OnStatefulSetChange(_ string, statefulSet *appsv1.StatefulSet)
 	return statefulSet, err
 }
 
+func (h *handler) OnPvcChange(_ string, pvc *corev1.PersistentVolumeClaim) (*corev1.PersistentVolumeClaim, error) {
+	if pvc == nil || pvc.DeletionTimestamp != nil || pvc.Namespace != util.HarvesterSystemNamespaceName {
+		return pvc, nil
+	}
+
+	// We only care about the log-archive PVC created by the fluentd StatefulSet
+	pvcLabels := labels.Set(pvc.Labels)
+	if !logArchiveMatchingLabels.AsSelector().Matches(pvcLabels) {
+		return pvc, nil
+	}
+	upgradeLogName, ok := pvc.Labels[util.LabelUpgradeLog]
+	if !ok {
+		return pvc, nil
+	}
+
+	upgradeLog, err := h.upgradeLogCache.Get(util.HarvesterSystemNamespaceName, upgradeLogName)
+	if err != nil {
```

Comment:
shall we skip the none-existing error?

if the pvc accidently has such a label